src/interfaces/node.cpp

```diff
@@ -228,6 +229,32 @@ class NodeImpl : public Node
         return wallets;
 #else
         throw std::logic_error("Node::getWallets() called in non-wallet build.");
+#endif
+    }
+    std::pair<std::unique_ptr<Wallet>, std::string> loadWallet(const std::string& wallet_file) override
+    {
+#ifdef ENABLE_WALLET
+        try {
+            std::string dummy_warning;
+            std::string error;
+
+            if (!CWallet::Verify(wallet_file, false, error, dummy_warning)) {
+                return { std::unique_ptr<Wallet>(), error };
+            }
+
+            CWallet * const wallet = CWallet::CreateWalletFromFile(wallet_file, fs::absolute(wallet_file, GetWalletDir()));
+            if (!wallet) {
+                return { std::unique_ptr<Wallet>(), "Wallet creation failed." };
+            }
+            AddWallet(wallet);
+
+            wallet->postInitProcess();
+            return { MakeWallet(*wallet), std::string() };
+        } catch (...) {
+            return { std::unique_ptr<Wallet>(), "ops" };
```

Comment:
I don't understand why you're passing back an error string if it's always the string `ops`

Reply:
Removed, change the return type.